app/controllers/houseConsignment/index/items/packages/RemovePackageTypeYesNoController.scala

```diff
@@ -0,0 +1,110 @@
+    extends FrontendBaseController
+    with I18nSupport {
+
+  private def addAnother(arrivalId: ArrivalId, mode: Mode): Call =
+    Call("GET", "#") //TODO should go to addAnother package page
+
+  private type Request = SpecificDataRequestProvider1[PackageType]#SpecificDataRequest[_]
+
+  private def packageType(implicit request: Request): PackageType = request.arg
+
+  private def form(packageType: PackageType): Form[Boolean] =
+    formProvider("houseConsignment.index.items.packages.removePackageTypeYesNo", packageType.toString)
+
+  def onPageLoad(arrivalId: ArrivalId, houseConsignmentIndex: Index, itemIndex: Index, packageIndex: Index, mode: Mode): Action[AnyContent] =
+    actions
+      .getStatus(arrivalId)
+      .andThen(getMandatoryPage(PackageTypePage(houseConsignmentIndex, itemIndex, packageIndex))) {
+        implicit request =>
+          val quantity = request.userAnswers.get(NumberOfPackagesPage(houseConsignmentIndex, itemIndex, packageIndex)).map(_.toString())
+          Ok(
+            view(form(packageType), request.userAnswers.mrn, arrivalId, houseConsignmentIndex, itemIndex, packageIndex, mode, insetText(quantity, packageType))
+          )
+      }
+
+  def onSubmit(arrivalId: ArrivalId, houseConsignmentIndex: Index, itemIndex: Index, packageIndex: Index, mode: Mode): Action[AnyContent] =
+    actions
+      .getStatus(arrivalId)
+      .andThen(getMandatoryPage(PackageTypePage(houseConsignmentIndex, itemIndex, packageIndex)))
+      .async {
+        implicit request =>
+          val quantity = request.userAnswers.get(NumberOfPackagesPage(houseConsignmentIndex, itemIndex, packageIndex)).map(_.toString())
+          form(packageType)
+            .bindFromRequest()
+            .fold(
+              formWithErrors =>
+                Future.successful(
+                  BadRequest(
+                    view(formWithErrors,
+                         request.userAnswers.mrn,
+                         arrivalId,
+                         houseConsignmentIndex,
+                         itemIndex,
+                         packageIndex,
+                         mode,
+                         insetText(quantity, packageType)
+                    )
+                  )
+                ),
+              value =>
+                for {
+                  updatedAnswers <-
+                    if (value) {
+                      Future.fromTry(request.userAnswers.removeExceptSequenceNumber(PackagingSection(houseConsignmentIndex, itemIndex, packageIndex)))
+                    } else { Future.successful(request.userAnswers) }
+                  _ <- sessionRepository.set(updatedAnswers)
+                } yield Redirect(addAnother(arrivalId, mode))
+            )
+      }
+
+  private def insetText(quantity: Option[String], packageType: PackageType): Option[String] =
+    quantity
+      .map(
+        value => s"$value ${packageType.toString}"
+      )
+      .orElse(Some(packageType.toString))
```

Comment:
`.orElse(Some(packageType.toString))` = `.getOrElse(packageType.toString)`

Reply:
We use the Optionality in the View additionalHtml property, hence why I have gone with the above approach @jackheslop96 .

Reply:
I will look at moving the optionality logic to the view instead.